app/test/fuzz_abci_test.go

```diff
@@ -2,52 +2,51 @@ package app_test
 	"github.com/celestiaorg/celestia-app/testutil"
-	"github.com/celestiaorg/celestia-app/x/blob/types"
+	"github.com/celestiaorg/celestia-app/testutil/blobfactory"
 	"github.com/stretchr/testify/require"
 	abci "github.com/tendermint/tendermint/abci/types"
-	tmrand "github.com/tendermint/tendermint/libs/rand"
 	core "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/types"
 )
 
-// TestFuzzPrepareProcessProposal produces blocks with random data using
+// TestPrepareProposalConsistency produces blocks with random data using
 // PrepareProposal and then tests those blocks by calling ProcessProposal. All
 // blocks produced by PrepareProposal should be accepted by ProcessProposal. It
 // doesn't use the standard go tools for fuzzing as those tools only support
-// fuzzing limited types, which forces us to create random block data ourselves
-// anyway. We also want to run this test alongside the other tests and not just
-// when fuzzing.
-func TestFuzzPrepareProcessProposal(t *testing.T) {
+// fuzzing limited types, instead we create blocks our selves using random
+// transction.
+func TestPrepareProposalConsistency(t *testing.T) {
 	encConf := encoding.MakeConfig(app.ModuleEncodingRegisters...)
-	signer := types.GenerateKeyringSigner(t, types.TestAccName)
-	testApp := testutil.SetupTestAppWithGenesisValSet(t)
-	timer := time.After(time.Second * 30)
-	for {
-		select {
-		case <-timer:
-			return
-		default:
-			t.Run("randomized inputs to Prepare and Process Proposal", func(t *testing.T) {
-				pfbTxs := app.GenerateManyRawWirePFB(t, encConf.TxConfig, signer, tmrand.Intn(100), -1)
-				txs := app.GenerateManyRawSendTxs(t, encConf.TxConfig, signer, tmrand.Intn(20))
-				txs = append(txs, pfbTxs...)
-				resp := testApp.PrepareProposal(abci.RequestPrepareProposal{
-					BlockData: &core.Data{
-						Txs: txs,
-					},
-				})
-				res := testApp.ProcessProposal(abci.RequestProcessProposal{
-					BlockData: resp.BlockData,
-					Header: core.Header{
-						DataHash: resp.BlockData.Hash,
-					},
-				})
-				require.Equal(t, abci.ResponseProcessProposal_ACCEPT, res.Result)
-			})
-		}
+	testApp, _ := testutil.SetupTestAppWithGenesisValSet()
+	for i := 0; i < 200; i++ {
+		ProcessRandomProposal(t, 100, 50000, encConf, testApp)
 	}
 }
+
+func ProcessRandomProposal(
+	t *testing.T,
+	count,
+	maxSize int,
+	cfg encoding.Config,
+	capp *app.App,
+) {
+	txs := blobfactory.RandBlobTxsRandomlySized(cfg.TxConfig.TxEncoder(), count, maxSize)
+	sendTxs := blobfactory.GenerateManyRawSendTxs(cfg.TxConfig, count)
+	txs = append(txs, sendTxs...)
+	resp := capp.PrepareProposal(abci.RequestPrepareProposal{
+		BlockData: &core.Data{
+			Txs: coretypes.Txs(txs).ToSliceOfBytes(),
+		},
+	})
+	res := capp.ProcessProposal(abci.RequestProcessProposal{
+		BlockData: resp.BlockData,
+		Header: core.Header{
+			DataHash: resp.BlockData.Hash,
+		},
+	})
+	require.Equal(t, abci.ResponseProcessProposal_ACCEPT, res.Result)
+}
```

Comment:
I've ran this locally for roughly an hour with no errors. Due to the simpler square estimation, square writing, and removal of the malleation process, I'm extremely hopeful that we will have a meaningful improvement in stability.j

however, due to how the transactions are generated and relatively large amount of compute required for this, the test does take a while. we might want to consider only running this test when we are not using the short flag.

Reply:
> when we are not using the short flag.

+1

Related https://github.com/celestiaorg/celestia-app/issues/1025